### PR TITLE
Fix for #577 (checkAssertions checks contract address) & more descriptive variable names & comments

### PR DESCRIPTION
**Pull request by snd**
tests pass for me locally.
unfortunately the CI doesn't run for this PR. probably because it doesn't merge into master.

fixes https://github.com/crytic/echidna/issues/577.

thanks @ggrieco-tob for helping me out with https://github.com/crytic/echidna/pull/679/commits/90392fb4bf772114e30c8e3d6757466a9a759b57.

### CODE REVIEW

**Comment by ggrieco-tob** [comment]
can you sync with the latest changes in `dev-test-refactoring` ? (the tests were refactored)

**Comment by snd** [comment]
@ggrieco-tob I already did. I rebased on dev-test-refactoring and fixed the conflicts before creating this PR.

**Review by ggrieco-tob — APPROVED** [review]

**Review by arcz — COMMENTED** [review]
Really nice code, thanks!

File: lib/Echidna/Test.hs [hunk 1]
```diff
@@ -156,20 +156,23 @@ checkOptimization em (f,a) = do
...
+      isCorrectAddr = addr == vm ^. state . codeContract
+      isCorrectTarget = isCorrectFn && isCorrectAddr
+      -- Whether the last transaction executed opcode 0xfe, meaning an assertion failure.
+      isAssertionFailure = case vm ^. result of
+        Just (VMFailure (UnrecognizedOpcode 0xfe)) -> True
+        _ -> False
+      -- Test always passes if it doesn't target the last executed contract and function.
+      -- Otherwise it passes if it doesn't cause an assertion failure.
+      isSuccess = not isCorrectTarget || not isAssertionFailure
+  pure (BoolValue $ isSuccess, extractEvents eventMap vm, getResultFromVM vm)
```
[the remark this answers is not in the document]
Comment (ggrieco-tob) [reply lib/Echidna/Test.hs:175]: It is not. I already fixed that on the echidna 2.0 branch

**Pull request merged**